Source/Core/Common/StringUtil.cpp

```diff
@@ -391,17 +407,21 @@ std::vector<std::string> SplitString(const std::string& str, const char delim)
 
 std::string JoinStrings(const std::vector<std::string>& strings, const std::string& delimiter)
 {
-  // Check if we can return early, just for speed
+  std::string joined;
   if (strings.empty())
-    return "";
-
-  std::ostringstream res;
-  std::copy(strings.begin(), strings.end(),
-            std::ostream_iterator<std::string>(res, delimiter.c_str()));
-
-  // Drop the trailing delimiter.
-  std::string joined = res.str();
-  return joined.substr(0, joined.length() - delimiter.length());
+    return joined;
+  const auto iter_begin = strings.begin(), iter_end = strings.end();
+  std::size_t reserve = delimiter.size() * (strings.size() - !delimiter.empty());
+  for (auto iter = iter_begin; iter != iter_end; ++iter)
+    reserve += iter->size();
+  joined.reserve(reserve);
+  for (auto iter = iter_begin;;)
+  {
+    joined += *iter;
+    if (++iter == iter_end)
+      return joined;
+    joined += delimiter;
+  }
```

Comment:
I think this entire function can be replaced with `fmt::format("{}", fmt::join(strings, delimiter));`

Reply:
Oh, that function sounds fun.  There was a bunch of other places where non-strings were being concatenated with delimiters that I saw but couldn't do anything about.  Full steam ahead on {fmt} dependencies, then?